### ISSUE TITLE
error in _ae_loss

### ISSUE THREAD

**Issue by VICKY1991** [zzzxxxttt/pytorch_simple_cornernet#2]
pull_loss, push_loss = _ae_loss(embd_tl, embd_br, batch['ind_masks'])
  File "/home/sambit/PROGRAMMING/pytorch_simple_CornerNet-master/utils/losses.py", line 84, in _ae_loss
    dist = F.relu(1 - (embd_mean[:, None, :] - embd_mean[:, :, None]).abs(), inplace=True)
IndexError: too many indices for tensor of dimension 1

**Comment by zzzxxxttt** [comment]
What is your batchsize? The ae loss function is incompatible with batchsize==1.

**Comment by PRITISHA2105** [comment]
Thank you so much for the feedback. This is indeed my batch size. (1)
Unfortunately, I have only a RTX2080 GPU, with 8GB.
So, I found that batch size > 1 causes out of memory error.
Just a question - the forward + backward has a memory foot print > 7GB.
Does it mean only the training phase needs > 8GB memory and final trained weights are about 800MB?
Thank you again.

**Comment by zzzxxxttt** [comment]
@PRITISHA2105 Yes, training requires more memory than evaluation, try to use Hourglass-52 ("small hourglass") as backbone if you don't have enough memory.
